src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] currentList;
+
+    public ArrayList() {
+        currentList = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        currentList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexChecking(index);
+        if (size == currentList.length) {
+            growIfArrayFull();
+        }
 
+        Object[] newList = new Object[currentList.length];
+        System.arraycopy(currentList, 0, newList, 0, index);
+        newList[index] = value;
+        System.arraycopy(currentList, index, newList, index + 1, size - index);
```

Comment:
in all methods of the class, use only one System.arraycopy - hint: you can have the same array as both src and dest arguments passed to it